spec/std/json/pull_parser_spec.cr

```diff
@@ -358,13 +366,67 @@ describe JSON::PullParser do
       end
     {% end %}
 
+    it_reads Float32::MIN
+    it_reads -10_f32
+    it_reads 0_f32
+    it_reads 10_f32
+    it_reads Float32::MAX
+
+    it_reads Float64::MIN
+    it_reads -10_f64
+    it_reads 0_f64
+    it_reads 10_f64
+    it_reads Float64::MAX
+
+    it_reads Int64::MIN
+    it_reads -10_i64
+    it_reads 0_i64
+    it_reads 10_i64
+    it_reads Int64::MAX
+
+    it "reads > Int64::MAX" do
+      pull = JSON::PullParser.new(Int64::MAX.to_s + "0")
+      pull.read?(Int64).should be_nil
+    end
+
+    it "reads < Int64::MIN" do
+      pull = JSON::PullParser.new(Int64::MIN.to_s + "0")
+      pull.read?(Int64).should be_nil
+    end
+
+    it "reads > Float32::MAX" do
+      pull = JSON::PullParser.new(Float64::MAX.to_s)
```

Comment:
Sorry if I'm missing something, but is this also supposed to have the ` + "0"` part?

Reply:
I think you're missing that the tested value is `Float64::MAX`, not `Float32::MAX`.

Reply:
Sorry; I just now saw this; thanks. The appended `0` for the `reads > Int64::MAX` test makes sense now; that would make the `1.7976931348623157e+308` value into `1.7976931348623157e+3080`, which would be more positive. Likewise, for the `reads < Int64::MIN` test, that would make the `-1.7976931348623157e+308` value into `-1.7976931348623157e+3080`, which would be more negative.